tests/layer_validation_tests.h

```diff
@@ -270,6 +270,24 @@ class VkLayerTest : public VkRenderFramework {
         return GetPhysicalDeviceProperties2(props2);
     }
 
+    template <typename Proc>
+    const Proc GetInstanceProcAddr(const char *proc_name, bool assert_proc = true) {
+        auto proc = reinterpret_cast<Proc>(vk::GetInstanceProcAddr(instance(), proc_name));
+        if (assert_proc) {
+            assert(proc);
+        }
+        return proc;
+    }
+
+    template <typename Proc>
+    const Proc GetDeviceProcAddr(const char *proc_name, bool assert_proc = true) {
+        auto proc = reinterpret_cast<Proc>(vk::GetDeviceProcAddr(device(), proc_name));
```

Comment:
Ensure the template parameter is a function.

```suggestion
        static_assert(std::is_function_v<Proc>);
        auto proc = reinterpret_cast<Proc>(vk::GetDeviceProcAddr(device(), proc_name));
```

Reply:
Oh I did not know about that, will do